extensions/objects/templates/filepath-editor.component.spec.ts

```diff
@@ -0,0 +1,2312 @@
+      .toBe('ew-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_INSIDE])
+      .toBe('move');
+    expect(component.data).toEqual({
+      mode: component.MODE_EMPTY, metadata: {}, crop: true });
+    expect(component.imageResizeRatio).toBe(1);
+    expect(component.cropArea).toEqual({ x1: 0, y1: 0, x2: 0, y2: 0 });
+    expect(component.mousePositionWithinCropArea).toBe(null);
+    expect(component.mouseLastKnownCoordinates).toEqual({ x: 0, y: 0 });
+    expect(component.lastMouseDownEventCoordinates).toEqual({ x: 0, y: 0 });
+    expect(component.userIsDraggingCropArea).toBe(false);
+    expect(component.userIsResizingCropArea).toBe(false);
+    expect(component.cropAreaResizeDirection).toBe(null);
+    expect(component.invalidTagsAndAttributes).toEqual({tags: [], attrs: []});
+    expect(component.processedImageIsTooLarge).toBe(false);
+    expect(component.entityId).toBe('2');
+    expect(component.entityType).toBe('question');
+  });
+
+  it('should change on mouseup event', () => {
+    component.userIsDraggingCropArea = true;
+    component.userIsResizingCropArea = true;
+
+    window.dispatchEvent(new Event('mouseup'));
+
+    expect(component.userIsDraggingCropArea).toBe(false);
+    expect(component.userIsResizingCropArea).toBe(false);
+  });
+
+  it('should reset component', () => {
+    component.value = 'file_1';
+    spyOn(imagePreloaderService, 'getDimensionsOfImage')
+      .and.returnValue(dimensionsOfImage);
+    spyOn(component, 'setSavedImageFilename');
+
+    component.ngOnInit();
+
+    expect(component.setSavedImageFilename)
+      .toHaveBeenCalledWith('file_1', false);
+    expect(imagePreloaderService.getDimensionsOfImage)
+      .toHaveBeenCalledWith('file_1');
+    expect(component.imageContainerStyle).toEqual({
+      height: '350px',
+      width: '450px'
+    });
+  });
+
+  it('should validate that the Image file is saved', () => {
+    component.data.mode = 3;
+
+    expect(component.validate(component.data)).toBe(true);
+  });
+
+  it('should reset file path editor and delete file if present', () => {
+    spyOn(contextService, 'getImageSaveDestination').and.returnValue(
+      AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE);
+    spyOn(imageLocalStorageService, 'isInStorage').and.returnValue(true);
+    spyOn(imageLocalStorageService, 'deleteImage');
+
+    expect(component.data.mode).toBe(2);
+    expect(component.data.metadata).not.toEqual({});
+    expect(component.data.crop).toBe(false);
+
+    component.resetFilePathEditor();
+
+    expect(imageLocalStorageService.deleteImage).toHaveBeenCalledWith(
+      'saved_file_name.png'
+    );
+    expect(component.data.mode).toBe(1);
+    expect(component.data.metadata).toEqual({});
+    expect(component.data.crop).toBe(true);
+    expect(component.imageResizeRatio).toBe(1);
+    expect(component.invalidTagsAndAttributes).toEqual({
+      tags: [],
+      attrs: []
+    });
+  });
+
+  it('should reset file path editor', () => {
```

Comment:
ditto comment, for here and elsewhere -- try to follow the `should do X when Y happens`

Reply:
Done